great_expectations/rule_based_profiler/types/data_assistant_result/plot_util.py

```diff
@@ -0,0 +1,101 @@
+from abc import ABC, abstractmethod, abstractproperty
+from dataclasses import dataclass
+from typing import Optional, Union
+
+import altair as alt
+
+
+class PlotUtil(ABC):
+    @abstractproperty
+    def title(self) -> str:
+        pass
+
+    @abstractmethod
+    def generate_tooltip(self) -> alt.Tooltip:
+        pass
+
+    @abstractmethod
+    def plot_on_target_axis(
+        self, axis: Union[alt.X, alt.Y], shorthand: Optional[str] = None
+    ) -> Union[alt.X, alt.Y]:
+        pass
+
+
+@dataclass(frozen=True)
+class MetricPlotUtil(PlotUtil):
+    metric_name: str
+    metric_type: alt.StandardType
+
+    @property
+    def title(self) -> str:
+        return self.metric_name.replace("_", " ").title()
+
+    def generate_tooltip(self) -> alt.Tooltip:
+        return alt.Tooltip(
+            field=self.metric_name,
+            type=self.metric_type,
+            title=self.title,
+            format=",",
+        )
+
+    def plot_on_target_axis(
+        self, axis: Union[alt.X, alt.Y], shorthand: Optional[str] = None
+    ) -> Union[alt.X, alt.Y]:
+        if shorthand is None:
+            shorthand = self.metric_name
+        return axis(
+            shorthand,
+            type=self.metric_type,
+            title=self.title,
+        )
+
+
+@dataclass(frozen=True)
+class DomainPlotUtil(PlotUtil):
+    domain_name: str
+    domain_type: alt.StandardType
+
+    @property
+    def title(self) -> str:
+        return self.domain_name.title()
+
+    def generate_tooltip(self) -> alt.Tooltip:
+        return alt.Tooltip(
+            field=self.domain_name,
+            type=self.domain_type,
+            title=self.title,
+            format=",",
+        )
+
+    def plot_on_target_axis(
+        self, axis: Union[alt.X, alt.Y], shorthand: Optional[str] = None
+    ) -> Union[alt.X, alt.Y]:
+        if shorthand is None:
+            shorthand = self.domain_name
+        return axis(
+            shorthand,
+            type=self.domain_type,
+            title=self.title,
+        )
+
+
+@dataclass(frozen=True)
+class BatchIdPlotUtil(PlotUtil):
```

Comment:
Pros/cons of dataclasses here? These were originally larger value bags but it may make less sense now?

Reply:
I'm happy with the dataclasses since it seems to clean up and simplify `DataAssistantResult`. They would probably be fine in the `data_assistant_result.py` module at the bottom though.

Reply:
I'm erring on the side of a separate file just in case the types expand but let me know if you feel strongly about consolidation.